### PR TITLE
Theme integration

### PR DESCRIPTION
**Pull request by hetmann**
- added `withGalio`, `GalioProvider`, `GalioTheme`

### CODE REVIEW

File: src/theme/sizes.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  BUTTON_WIDTH: 16 * 9,
+  BUTTON_HEIGHT: 16 * 2.75,
+  BUTTON_SHADOW_RADIUS: 10,
+
+  // Block styles
+  BLOCK_SHADOW_OPACITY: 0.15,
+  BLOCK_SHADOW_RADIUS: 8,
+  CARD_BORDER_RADIUS: 16 * 0.4,
+  CARD_BORDER_WIDTH: 16 * 0.05,
+  ELEVATION: 1,
```
Comment (paullaros) [inline src/theme/sizes.js]: Hard to guess what this one is doing

File: src/theme/sizes.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  // Button styles
+  BUTTON_WIDTH: 16 * 9,
+  BUTTON_HEIGHT: 16 * 2.75,
+  BUTTON_SHADOW_RADIUS: 10,
+
+  // Block styles
+  BLOCK_SHADOW_OPACITY: 0.15,
+  BLOCK_SHADOW_RADIUS: 8,
+  CARD_BORDER_RADIUS: 16 * 0.4,
```
Comment (paullaros) [inline src/theme/sizes.js]: Move this one?

File: src/theme/sizes.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  CARD_BORDER_WIDTH: 16 * 0.05,
```
Comment (paullaros) [inline src/theme/sizes.js]: Move this one?

File: src/theme/sizes.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  INPUT_HELP_TEXT: 16 * 0.8,
+  INPUT_ROUNDED: 16 * 1.7,
+
+  // NavBar styles
+  NAVBAR_HEIGHT: 16 * 4.125,
+  NAVBAR_VERTICAL: 16,
+  NAVBAR_TITLE_FLEX: 2,
+  NAVBAR_TITLE_HEIGHT: height * 0.07,
+  NAVBAR_TITLE_TEXT: 16 * 0.875,
+  NAVBAR_LEFT_FLEX: 0.5,
```
Comment (paullaros) [inline src/theme/sizes.js]: Combine NAVBAR_LEFT_* & NAVBAR_RIGHT_* as they are 'always' the same?

**Comment by paullaros** [comment]
Neat! Really like the naming conventions you're using.

File: src/theme/sizes.js (hunk 1 above)
Comment (hetmann) [reply src/theme/sizes.js]: Elevation is Android specific :)

File: src/theme/sizes.js (hunk 4 above)
Comment (hetmann) [reply src/theme/sizes.js]: Are the same at this point, for better customisation anyone can change the value using directly this constants

Comment (hetmann) [reply src/theme/sizes.js]: @paullaros @palingheorghe please check the implementation of the theme & components.

Comment (hetmann) [reply src/theme/sizes.js]: @paullaros @palingheorghe idea: add theme as separate npm package, e.g.: galio-theme
What do you think?

Comment (hetmann) [reply src/theme/sizes.js]: @paullaros @palingheorghe 
- **galio-framework** is at version **1.1.0**
- **expo** version **31.0.4**
- **react** version **16.5.0**

**examples** branch will need to upgrade to version newest expo version

**Comment by palingheorghe** [comment]
Hi,
I encountered some problems while testing out this in a different app.

**How I found out about the problems:**
1. I initialized another Expo app. (React v16.5, Expo v31.0.2)
2. I added all the necessary files from the 'theme' branch ( I basically transferred the 'src' folder and renamed it in 'components' )
3. in my `App.js` I imported `GalioProvider` and `withGalio`.
4. I have now tried several different ways of playing around with these.

**Using GalioProvider**
From my understanding this component's behavior should be the following: 
- The children of GalioProvider should have acces to our default theme through Context
- I should be able to pass down my own theme through the Provider 

Right now I cannot find any way of configuring the default theme (or creating a completly different theme) and then pass down my new or modified theme through the Provider. (The Provider doesn't accept any new value as it's using a static variable pre-defined in the theme/index.js file)

**Using withGalio**
The behavior I was expecting: 
- The component I'm using with `withGalio` should be able to acces my theme through the `styles` variable
```js
const styles = theme => StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.COLORS.FACEBOOK
  }
});
export default withGalio(App, styles);
```

This is working ✅ . My bgColor turned blue using the variable like this in a View component.
```js
<View style={this.props.styles.container}>
```

I think we need to better define the purpose of both functions and maybe create some sort of config function in order to configurate the already existent theme or create a new one. 
The only way of creating a new theme, right now as far as I see, is to go directly into the theme folder and modify the existing file which is not the best UX for people who want to use Galio as their main UI library.

Thanks! 🤙🏽

File: package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "galio-framework",
   "main": "src/index.js",
-  "version": "1.0.3",
+  "version": "1.1.0",
```
Comment (palingheorghe) [inline package.json]: We were thinking about this and we're actually going to change the version to 0.5 until we create more components, screens and just a better UX for our users.

**Comment by palingheorghe** [comment]
✅Both 'withGalio' and 'GalioProvider' functions work correctly now. 

We just have to change the `package.json` file (more about this above) and we're good to go, at least from my perspective.

Thanks! ❤️🤙🏽

**Comment by hetmann** [comment]
> ✅Both 'withGalio' and 'GalioProvider' functions work correctly now.
> 
> We'll just have to change the `package.json` file (more about this above) and we're good to go, at least from my perspective.
> 
> Thanks! ❤️🤙🏽

Just updated to version 1.1.1 that fixed the custom theme pass from **GalioProvider**
To use it install the latest version `npm install galio-framework@latest` or `yarn add galio-framework@latest`

**Comment by antelbra** [comment]
Amazing work @hetmann 👌

**Comment by hetmann** [comment]
Just to inform you guys & girls :) I'm writing a small README for the theme to understand how to use it & customise your components. I'll publish the updates in no time :)

File: package.json (hunk 5 above)
Comment (palingheorghe) [reply package.json]: This was the ‘above comment’ I was mentioning @hetmann

**Comment by palingheorghe** [comment]
This is it! We're now ready to merge this beautiful new feature into the **Alpha-1** branch. 

Good job everyone! Nice! 

🤙🏽

**Pull request merged**